lib/generators/manageiq/provider/provider_generator.rb

```diff
@@ -10,13 +10,32 @@ class ProviderGenerator < PluginGenerator
     class_option :dummy, :type => :boolean, :default => false,
                  :desc => "Generate dummy implementations (Default: --no-dummy)"
 
+    class_option :manager_type, :type => :string, :default => "cloud",
+                 :desc => "What type of manager to create (Default: \"cloud\")"
+
+    def self.manager_types
+      @manager_types ||= {
+        "automation"    => "AutomationManager",
+        "cloud"         => "CloudManager",
+        "configuration" => "ConfigurationManager",
+        "container"     => "ContainerManager",
+        "infra"         => "InfraManager",
+        "monitoring"    => "MonitoringManager",
```

Comment:
Do we still have monitoring managers?

Reply:
Yes, kubernetes and openshift have monitoring managers based on prometheus_alerts and we could add thanos as an OCPv4 monitoring manager